sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,264 @@
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationManual,
+      .reseed_on_key_change = true,
+      .force_masks = false,
+      .ctrl_aux_lock = false,
+  };
+
+  LOG_INFO(
+      "AES transaction configured: ECB mode, 128-bit key, manual operation.");
+
+  // Disable EDN and CSRNG to simulate stalling
+  LOG_INFO("Disabling EDN0, EDN1...");
+  CHECK_DIF_OK(dif_edn_stop(&edn0));
+  CHECK_DIF_OK(dif_edn_stop(&edn1));
+  busy_spin_micros(500);
+
+  LOG_INFO("Conditionally checking AES status after stopping EDN and CSRNG.");
+  LOG_INFO("Waiting for AES to stall after PRNG reseed request...");
+  bool stall = false;
+  CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusStall, &stall));
+
+  if (stall) {
+    LOG_INFO("AES encryption has stalled as expected.");
+  } else {
+    LOG_ERROR(
+        "ERROR:AES encryption did not stall as expected after EDN stop! Lets "
+        "proceed "
+        "for now to verify further steps...");
+  }
+
+  log_csrng_internal_state(&csrng);
+  monitor_entropy_sources();
+
+  // Expanding aes_testutils_setup_encryption T1 to T8:
+  // T: Start of aes_testutils_setup_encryption function steps
+  // T1: Mask the key. Note that this should not be done manually. Software is
+  // expected to get the key in two shares right from the beginning.
+  uint8_t key_share0[sizeof(kAesModesKey128)];
+  for (int i = 0; i < sizeof(kAesModesKey128); ++i) {
+    key_share0[i] = kAesModesKey128[i] ^ kKeyShare1[i];
+  }
+
+  // T2: "Convert" key share byte arrays to `dif_aes_key_share_t`.
+  dif_aes_key_share_t key;
+  memcpy(key.share0, key_share0, sizeof(key.share0));
+  memcpy(key.share1, kKeyShare1, sizeof(key.share1));
+
+  // T3:  Wait for status kDifAesStatusIdle
+  // Start AES with the given transaction
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusIdle, true, kTestTimeout);
+
+  // T4: dif_aes_start
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, NULL));
+
+  LOG_INFO("Disabling CSRNG...");
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
+  CHECK_DIF_OK(dif_entropy_src_stop(&entropy_src));
+
+  // Re-enable Entropy Source and CSRNG in the proper sequence
+  LOG_INFO("Re-initializing Entropy Source and CSRNG...");
+
+  // Initialize and configure Entropy Source
+  dif_entropy_src_config_t entropy_src_config =
+      entropy_testutils_config_default();
+  CHECK_DIF_OK(dif_entropy_src_configure(&entropy_src, entropy_src_config,
+                                         kDifToggleEnabled));
+
+  // Initialize and start CSRNG
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+  // set_edn_auto_mode();
+  LOG_INFO("Setting EDN to auto mode...");
+  TRY(entropy_testutils_auto_mode_init());
+  // CHECK_DIF_OK(entropy_testutils_auto_mode_init());
+  monitor_edn_status(&edn0, &edn1);
+
+  // Trigger PRNG reseed request
+  LOG_INFO("Triggering PRNG reseed...");
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerPrngReseed));
```

Comment:
I would expect to see a stall if entropy complex is disabled and you call this command. It may be a good idea to test this as well.

Reply:
Old comment